src/udev/net/ethtool-util.c

```diff
@@ -206,3 +212,116 @@ int ethtool_set_wol(int *fd, const char *ifname, WakeOnLan wol) {
 
         return 0;
 }
+
+static int ethtool_get_stringset(int *fd, struct ifreq *ifr, int stringset_id, struct ethtool_gstrings **gstrings) {
+        _cleanup_free_ struct ethtool_sset_info *info = NULL;
+        _cleanup_free_ struct ethtool_gstrings *strings = NULL;
+        unsigned len;
+        int r;
+
+        info = malloc0(sizeof (*info) + sizeof (unsigned));
```

Comment:
Given that the size is const you might as well allocate this directly, by defining:

```c
struct {
        struct ethtool_sset_info info;
        unsigned foo;
} buffer =  {
        .cmd = ETHTOOL_GSSET_INFO;
        …
};
```

That saves you the malloc handling, and is faster...